lib/pure/math.nim

```diff
@@ -100,7 +100,12 @@ proc nextPowerOfTwo*(x: int): int {.noSideEffect.} =
   result = result or (result shr 4)
   result = result or (result shr 2)
   result = result or (result shr 1)
-  result += 1 + ord(x<=0)
+  result += 1 
+  # Handle the case where x is higher than the greatest power of two.
```

Comment:
meh, that's more complex than your previous solution. If it's that complex, better leave it working on ``int``?